numpy/_core/arrayprint.py

```diff
@@ -85,7 +85,13 @@ def _make_options_dict(precision=None, threshold=None, edgeitems=None,
     if sign not in [None, '-', '+', ' ']:
         raise ValueError("sign option must be one of ' ', '+', or '-'")
 
-    if legacy == False:
+    if legacy is False:
+        options['legacy'] = sys.maxsize
+    elif legacy == False:  # noqa: E712
+        warnings.warn(
+            f"Passing `legacy={legacy}` is deprecated.",
```

Comment:
```suggestion
            f"Passing `legacy={legacy!r}` is deprecated.",
```

Just in case. It's almost certainly going to be `0`, but who knows?